### PR TITLE
Fix address ui

### LINKED ISSUE #15: Update Address overview to fit on mobile
As a user, I want the address overview to be responsive, so that I can view it on my mobile. 

# Replication steps 

 1. In mobile view, click ‘Explore’ > ‘Transaction’

 2. Then click an address link in the `From` or To `column`

The address overview page has images overlapping with text and poor formatting. Please update it, to make it match the attached design.

# Prerequisites

You will need to know basic Javascript (jQuery) and css (scss) to do this task. The knowledge of nunjucks will be helpful.

# Guidelines & Hints

For the Transactions section at the end of the page, we always want to show the scrollbar, so the user knows in advance that there is more data to be seen. 

You can check the code used in the Menu -> Transactions page.

The address overview page is located at: views/pages/address-detail.njk.

The current scrollbar is implemented on views/pages/tx-list.njk (look for className scroll-bar-wrapper).

The code to handle the scrollbar is located at: src/assets/js/mobile/scroll-bar.js (you should not have to change this).

Add any other necessary mobile css to _mobile.scss to match the design.

Please make sure your code doesn’t affect the desktop version. 

We do not want to change the HTML structure. 

We also recommend putting all mobile related styling code to the `_mobile.scss` file. 

# Resources

## Repository link

https://github.com/cennznet/explorer/tree/master 

This ‘explorer’ repository contains several projects. 

The project you need to work on is in the ‘portal’ folder.

## Design:

Please refer to the InVision prototype here:  [UNcover Bounty Tasks](https://invis.io/7TUN9MQY259)

You will need to request access permission to access inspect mode on the InVision prototype. 

Screenshot of intended design also attached. 

*Please note there may be some inconsistencies with the InVision prototype and the live CSS. If this occurs, please use the styles defined by the exisiting CSS. 

## Get Up & Running 

Please check the README and CONTRIBUTING files:

https://github.com/cennznet/explorer/blob/master/portal/README.md https://github.com/cennznet/explorer/blob/master/portal/CONTRIBUTING.md 

# Test & Verify Your Changes 

## Mobile Devices 

Ensure your changes work on: 

- The latest Android and iPhones phones.
- Chrome, Safari and FireFox responsive views.

Ensure your changes work as expected and look exactly like the attached design. 

Also ensure that the changes do not introduce bugs or break the desktop version.

## Expected Outcome

**Given** user navigates to address overview page using mobile device

**When** user navigates to address section

**Then** user should see QR code with address

**And**  user should see full address

**And** user should see copy symbol

**Given** user navigates to address overview page using mobile device

**When** user navigates to address section

**And** user click the copy symbol

**Then** user should have address copied to the clipboard and a tooltip appears

**Given** user navigates to address overview page using mobile device

**When** user navigates to assets section

**Then** user should see all assets stored in that address

**Given** user navigates to address overview page using mobile device

**When** user navigates to assets section

**And** user click the buttons on the left top side of the assets section

**Then** user should see the assets displaying style changes

**Given** user navigates to address overview page using mobile device

**When** user navigates to transactions section

**Then** user should see time

 **And** user should see address

 **And** user should see incoming

 **And** user should see outgoing

 **And** user should see type

  **And** user should see status

  **And** user should see balance

**Given** user navigates to address overview page using mobile device

**When** user navigates to transactions section

 **And** user clicks on arrow sign at the end of the line

**Then** user should see expanded section

 **And** user should see block

 **And** user should see transaction

 **And** user should see  size 

 **And** user should see  fee

**Given** user navigates to address overview page using mobile device

**When** user navigates to transactions section

**Then** user should see transaction type selection dropdown

**Given** user navigates to address overview page using mobile device

**When** user navigates to transactions section

**Then** user should see transaction flow selection dropdown

**Given** user navigates to address overview page using mobile device

**When** user navigates to transactions section

**Then** user should see start date selection dropdown

**Given** user navigates to address overview page using mobile device

**When** user navigates to transactions section

**Then** user should see end date selection dropdown

# Current Screenshots

 
image

image

–––––––––––––––––––––––––––––––––––––––––––

# Designed Screen (how your design should look)
image

### PR DESCRIPTION
**Pull request by iamonuwa**
Fixes #15

### CODE REVIEW

File: portal/locales/en.json [hunk 1]
```diff
@@ -116,5 +116,68 @@
 	"User-generated": "User-generated",
 	"helper.field.show.mobile": "Show list of validators",
 	"helper.field.hide.mobile": "Hide list of validators",
-	"navbar.filter.transactions": "Filter Transactions"
+	"navbar.filter.transactions": "Filter Transactions",
+	"filter.token": "Token",
+	"filter.selectToken": "Select Token",
+	"21520891805000000000000": "21520891805000000000000",
```
Comment (centrality-bounties) [inline portal/locales/en.json]: what are these for? , i dont think they are needed

File: portal/src/assets/js/mobile/select.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+$(document).ready(() => {
+	let url = new URL(document.location);
+	let params = url.searchParams;
+	let asset = params.get('asset');
+
+	if (asset !== null) {
+		$('#txTokens').value = asset;
+	}
+
+	$('#txTokens').on('change', e => {
```
Comment (centrality-bounties) [inline portal/src/assets/js/mobile/select.js]: got a better name then e?

File: portal/src/assets/style/_mobile.scss [hunk 3]
```diff
@@ -112,13 +127,22 @@
     padding-right: 0px;
     padding-left: 0px;
     margin-top: 0px;
-    margin-bottom: 0px;
+	margin-bottom: 0px;
+	
+	.block-section {
+	  box-shadow: none;
+	  margin: 0;	
+	}
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: wrong formatting

File: portal/src/assets/style/_mobile.scss [hunk 4]
```diff
@@ -146,6 +170,12 @@
     padding: 5px 5px;
   }
 
+  .page-item .page-link {
+	color: #ffffff;
+	background-color: #1130FF;
+	font-weight: 600;
+	font-size: 0.8rem;
+  }
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: wrong formatting

File: portal/src/assets/style/_mobile.scss [hunk 5]
```diff
@@ -156,6 +186,10 @@
     border-top: none;
   }
 
+  .filter-form {
+	flex-direction: column;
+  }
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: wrong formatting

File: portal/src/assets/style/_mobile.scss [hunk 6]
```diff
@@ -273,6 +307,84 @@
...
+  
+	.qr-img {
+	  width: unset;
+	  order: 2;
+	  align-self: center;
+	  padding: 1.5rem 0;
+	  canvas {
+		width: 180px !important;
+		height: 180px !important;
+	  }
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: wrong formatting

File: portal/src/assets/style/_mobile.scss [hunk 7]
```diff
@@ -273,6 +307,84 @@
...
+	  width: unset;
+	  display: flex;
+	  order: 1;
+	  padding: 2rem 0.5rem 0 0.5rem;
+  
+	  h2 {
+	    padding: 0;
+		font-size: 1rem;
+		margin-right: 1rem;
+		font-weight: 600;
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: wrong formatting

File: portal/views/partials/address/address-tokens.njk [hunk 8]
```diff
@@ -71,3 +69,33 @@
...
+<div class="block-section address-tokens is-mobile">
+  <div class="block-section__header--cols">
+	<h2><strong>{{ _('filter.token') }}:</strong> ({{ address.balances.length }} {{ _('helper.field.total') }})</h2>
+  </div>
+  <div class="block-section__content filter-form">
+  <span>Select Token</span>
+	<select name="txTokens" id="txTokens" class="selectpicker" data-width="100%" data-size="5">
+      <option value="" selected>{{ _('filter.default.selectAll') }}</option>
+        {% for balanceInfo in address.balances %}
+          {% if balanceInfo.assetId === opts.token %}
```
Comment (centrality-bounties) [inline portal/views/partials/address/address-tokens.njk]: the Option block is repeated , find a way to add the selected property dynamically: maybe like:
<option value="" {{ balanceInfo.assetId === opts.token ? 'selected' :''}}">>
There should also only be one selected option

File: portal/src/assets/style/_mobile.scss (hunk 7 above)
Comment (iamonuwa) [reply portal/src/assets/style/_mobile.scss]: Hey, what's wrong with this formatting?

File: portal/src/assets/style/_mobile.scss (hunk 3 above)
Comment (iamonuwa) [reply portal/src/assets/style/_mobile.scss]: What's wrong with this formatting

**Comment by iamonuwa** [comment]
Made changes @centrality-bounties

File: portal/src/assets/style/_mobile.scss [hunk 9]
```diff
@@ -146,6 +171,14 @@
...
   }
 
+  .page-item {
+    .page-link {
+      color: #ffffff;
+      background-color: #1130FF;
+      font-weight: 600;
+      font-size: 0.8rem;
+	}
+  }
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: formatting : that curly brace in wrong place

File: portal/src/assets/style/_mobile.scss [hunk 10]
```diff
@@ -156,10 +189,16 @@
...
-  .filter-form .query-actions {
+  .filter-form {
     flex-direction: column;
   }
 
+  .filter-form {
+    .query-actions {
+      flex-direction: column;
+	}
+  }
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: formatting : that curly brace in wrong place

File: portal/src/assets/style/_mobile.scss [hunk 11]
```diff
@@ -273,6 +312,85 @@
...
+			color: #ffffff;
+			stroke-width: 30px;
+		  }
+		}
+		.btn {
+		  padding: 0;
+		}
+      }
+	}
+  }
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: formatting : that curly brace in wrong place

File: portal/src/assets/style/_mobile.scss [hunk 12]
```diff
@@ -273,6 +312,85 @@
...
+      padding-left: 0px;
+      margin-top: 0px;
+	  margin-bottom: 0px;
+	
+	  .block-section {
+	    border: 0;
+	    box-shadow: none;
+	    margin: 0;	
+	  }
+    }
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: formatting : that curly brace in wrong place

File: portal/src/assets/style/_mobile.scss [hunk 13]
```diff
@@ -408,6 +526,54 @@
   display: none;
 }
 @media only screen and (max-width: $mobile-width) {
+  .format-option {
+	display: flex;
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: formatting :you need indentation here

File: portal/src/assets/style/_mobile.scss [hunk 14]
```diff
@@ -408,6 +526,54 @@
...
+	flex-direction: row;
+	justify-content: space-between;
+	h6 {
+	  color: #1130FF;
+	  margin: 0;
+	}
+  }
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: formatting : that curly brace in wrong place

File: portal/src/assets/style/_mobile.scss [hunk 15]
```diff
@@ -408,6 +526,54 @@
...
+		text-align: center;
+		margin: 0;
+	  }
+	}
+	h2 {
+	  font-size: 1.5rem;
+	  text-align: left;
+	  margin-left: 1.5rem;
+	}
+  }
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: formatting : that curly brace in wrong place

File: portal/src/assets/style/_mobile.scss [hunk 16]
```diff
@@ -408,6 +526,54 @@
...
+	  padding: 1rem 1.5rem;
+	  span {
+	    font-size: 1rem;
+		  color: #4E5458;
+		}
+	}
+	.dropdown-item.active {
+	  background-color: #45C7D2;
+	}
+  }
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: formatting : that curly brace in wrong place

**Comment by centrality-bounties** [comment]
Two changes needed:

1. Make the copy icon fully appear , ( you can see it is slightly cut off on iphone ( maybe you just need to add or remove padding of something )

2. make the  asset drop down selected item  , the same color as the other ( transaction flow) drop down below. 
2. 1 Also align the asset name and asset amount the same way as the selected one . ( one to the left and one to the right

File: portal/src/assets/style/_mobile.scss [hunk 17]
```diff
@@ -538,22 +538,22 @@ hr {
   }
   .block-section {
     .block-section__header {
-	  padding: 0.5rem 0;
+      padding: 0.5rem 0;
 	}
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: this curly brace must  be indented

File: portal/locales/en.json [hunk 18]
```diff
@@ -116,5 +117,34 @@
...
+	"9899549445000000000000": "9899549445000000000000",
+	"9889547225000000000000": "9889547225000000000000",
+	"9889542785000000000000": "9889542785000000000000",
+	"9889533905000000000000": "9889533905000000000000",
+	"9889529465000000000000": "9889529465000000000000",
+	"99800000000000000000": "99800000000000000000",
+	"9889527245000000000000": "9889527245000000000000",
+	"153600000000000000000": "153600000000000000000",
+	"4720730000000000018": "4720730000000000018",
+	"976197202000000000000000": "976197202000000000000000"
```
Comment (centrality-bounties) [inline portal/locales/en.json]: Please remove this

File: portal/locales/en.json [hunk 19]
```diff
@@ -1,4 +1,5 @@
 {
+	"0": "0",
```
Comment (centrality-bounties) [inline portal/locales/en.json]: is this used ? if not please remove it

**Pull request closed without merging**